src/test/java/com/thealgorithms/sorts/BucketSortTest.java

```diff
@@ -7,42 +7,82 @@
 public class BucketSortTest {
```

Comment:
Why does it not use the common test base class for sorts?

Reply:
Because in that case we would have to remove all this code and provide a new one, since the current solution only supports int values.

Such test class:

```
public class BucketSortTest extends SortingAlgorithmTest {
    @Override
    SortAlgorithm getSortAlgorithm() {
        return new BucketSort();
    }
}
```
Require from us that our BucketSort implement this method `public <T extends Comparable<T>> T[] sort(T[] array);`

I will try to fix this. Common approach is much better and provides better testing.

Reply:
In fact, everything turned out to be simpler.